contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -12,8 +12,6 @@ with Langkit_Support.Diagnostics; use Langkit_Support.Diagnostics;
 with Langkit_Support.Diagnostics.Output;
```

Comment:
(commit message)
It’s not clear to me why this work on diagnostics is needed. Could you elaborate? Assuming this is about memory footprint: I woud have assumed that in the general case, there are mostly no diagnostics, and thus that their memory footprints was neglectible.

The biggest issue I see with this new implementation is that Diagnostic_Type instances (which exist in the public Ada API) have an unsafe reference to a TDH, yielding an unsafe API. We would need a safety net to fix this, probably cancelling the gain in memory footprint.

Reply:
This has been refactored to avoid this. Thanks because the result is much clearer anyway :)